commanduino/commanddevices/commandtcs34725.py

```diff
@@ -0,0 +1,143 @@
+"""
+
+.. module:: CommandTCS34725
+   :platform: Unix
+   :synopsis: Represents a TCS34725 Arduino Device.
+
+.. moduleauthor:: Artem Leonov <Artem.Leonov@glasgow.ac.uk>
+
+"""
+
+import time
+
+from .commanddevice import CommandDevice
+
+# Bonjour Information
+BONJOUR_ID = 'TCS34725'
+CLASS_NAME = 'CommandTCS34725'
+
+# Incoming
+CMD_READ_RGBC = 'R'
+CMD_REPLY_HEADER = 'C'
+CMD_INITIALIZE = 'Z'
+CMD_INITIALIZE_HEADER = 'E'
+
+# Outgoing
+CMD_SET_INTEGRATION_TIME = 'I'
+CMD_SET_GAIN = 'G'
+
+# Accepted values
+INTEGRATION_TIMES = [2.4, 24, 50, 101, 154, 700]
+GAINS = [1, 4, 16, 60]
+
+
+class CommandTCS34725(CommandDevice):
+    """
+    TCS34725 Arduino device.
+    
+    Implementation based on Adafruit #1334 RGB sensor.
+
+    Base:
+        CommandDevice
+    """
+    def __init__(self):
+        CommandDevice.__init__(self)
+        self.register_all_requests()
+
+        # value placeholders
+        self._integration_time = 0.05 # 50 ms by default
+        self._gain = 4 # default
+
+    def register_all_requests(self):
+        """ Registers all requests. """
+        self.register_request(
+            CMD_READ_RGBC,
+            CMD_REPLY_HEADER,
+            'rgbc',
+            self.handle_get_rgbc
+        )
+        self.register_request(
+            CMD_INITIALIZE,
+            CMD_INITIALIZE_HEADER,
+            'initialization_code',
+            self.handle_initialize,
+            timeout=1.2,
+        )
+
+    def handle_initialize(self, *arg):
+        """ Handles the sensor initialization command. """
+        if arg[0]:
+            self.initialization_code = int(arg[0])
+            self.initialization_code_lock.ensure_released()
+
+    def init(self):
+        """ Initializes the sensor. 
+        
+        Refer to self.initialization_code for checking if the initialization was successful.
+        """
+
+        self.get_initialization_code()
+
+        if self.initialization_code != 1:
```

Comment:
The documentation on the meaning of the initialization code could be added here rather than in the demo (where is less likely to be found)

Reply:
Added a note in the `init` docstring (165c4b2)